### ISSUE TITLE
how to tell stream is closed (by client)

### ISSUE THREAD

**Issue by cakir-enes** [elysiajs/elysia#341]
This just keeps looping and after handle doesn't trigger after I cancel the request.
```
app.get(
  '/:chain/:market/orders',
  async ({ isHX, optimizer, token0, token1, params: { chain, market } }) =>
    new Stream(async (stream) => {
      while (true) {
        console.log('sending')
        try {
          let c = stream.send(<Content />) // this is void..
          console.log('sent ', c)
        } catch (e) {
          console.log(e)
        }
        await Bun.sleep(1500)
      }
    }),
  {
    afterHandle(context) {
      console.log('after handle') // doenst run even when client closes conn
    },
  }
)
```

**Comment by feggaa** [comment]
This happen to you because use infinite loop " while (true) " 

You need to use Event to check if client close connection 
like webSocket
 ws.on("close", (event) => {
      // ....
  });
  
afterHandle hook is called before responding to the client and you cannot use it for your case

**Comment by Joxtacy** [comment]
I'm having what I think is the same question as the issue is asking.

I would like to know when the client disappears because I would like to do some cleanup after to close a rabbitmq consumer that I was using to send events on the stream. I haven't been able to figure out how to do that. Does anyone know? The lifecycle hooks all seem to trigger on the initial request to set up the sse.

**Comment by pg-wtatum** [comment]
I was able to detect an orderly shutdown of the request by looking at `ctx.request.signal.aborted` and it should be possible to use the events of the AbortSignal as well. 

See this sample which sends 15 messages but exits early if the client is not listening. Confirmed the early exit by watching server side for the console.log output
```
  .get(
    "/sse",
    (ctx) =>
      new Stream(async (stream) => {
        // Send up to 15 messages
        for (let i = 0; i < 15; i++) {
          // If the client has disconnected break out
          if (ctx.request.signal.aborted) {
            break;
          }
          console.log(`in the loop ${i}`);
          stream.send({
            i,
          });
          await stream.wait(500);
        }

        // Final message and close
        stream.send({
          final: true
        });
        stream.close();
      })
  )
  ```
  
  I have not been able to test yet how this behaves for an unorderly disconnect but I am **assuming** that Bun HTTP server will eventually react to client disconnect with a server-side timeout

**Comment by SaltyAom** [comment]
Request signal abort response has been implemented in 267ad71, available since 1.1.0.

Feel free to reopen the issue if the bug still persists.
